feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -371,18 +371,26 @@ bool target_radio_config_set2(const struct schema_Wifi_Radio_Config *rconf,
 
 	char phy[6];
 	char ifname[8];
+	bool is_csa_requested = false;
 
 	strncpy(ifname, rconf->if_name, sizeof(ifname));
 	strncpy(phy, target_map_ifname(ifname), sizeof(phy));
 
-	if (changed->channel && rconf->channel)
+	if (changed->channel && rconf->channel) {
 		blobmsg_add_u32(&b, "channel", rconf->channel);
+		is_csa_requested = true;
+	}
 
-	if (changed->enabled)
+	if (changed->enabled) {
 		blobmsg_add_u8(&b, "disabled", rconf->enabled ? 0 : 1);
+		is_csa_requested = false;
+	}
+
 
 	if (changed->tx_power) {
 		int max_tx_power;
+		is_csa_requested = false;
```

Comment:
This is incomplete.  If the the tx-power is set to greater than the max_power the AP will set it to max_power and a change will be reported continuously.  This change shouldn't block the channel rebalance operation.

A similar issue will occur with tx_chainmask, rx_chainmask and with ht_mode when set to Auto.

I think we are going to need my proposed change to track the Auto behaviour and shut down the change indication.  Chaitanya already included part of it in WIFI-2649 yesterday.

Will need to think about this.  I don't think this code needs to change - rather additional code needs to prevent these change flags from being set.